openshift-with-appstudio-test/e2e/jvm_build_service_test.go

```diff
@@ -4,265 +4,17 @@
-	// tests do now run in conjunction with say the full suite of e2e's in the e2e-tests runs, so no contention there.
-	//defer projectCleanup(ta)
-
-	path, err := os.Getwd()
-	if err != nil {
-		debugAndFailTest(ta, err.Error())
-	}
-	ta.Logf(fmt.Sprintf("current working dir: %s", path))
-
-	mavenYamlPath := filepath.Join(path, "..", "..", "deploy", "base", "maven-v0.2.yaml")
-	ta.maven = &v1beta1.Task{}
-	obj := streamFileYamlToTektonObj(mavenYamlPath, ta.maven, ta)
-	var ok bool
-	ta.maven, ok = obj.(*v1beta1.Task)
-	if !ok {
-		debugAndFailTest(ta, fmt.Sprintf("file %s did not produce a task: %#v", mavenYamlPath, obj))
-	}
-	// override images if need be
-	analyserImage := os.Getenv("JVM_BUILD_SERVICE_ANALYZER_IMAGE")
-	if len(analyserImage) > 0 {
-		ta.Logf(fmt.Sprintf("PR analyzer image: %s", analyserImage))
-		for _, step := range ta.maven.Spec.Steps {
-			if step.Name != "analyse-dependencies" {
-				continue
-			}
-			ta.Logf(fmt.Sprintf("Updating analyse-dependencies step with image %s", analyserImage))
-			step.Image = analyserImage
-		}
-	}
-	sidecarImage := os.Getenv("JVM_BUILD_SERVICE_SIDECAR_IMAGE")
-	if len(sidecarImage) > 0 {
-		ta.Logf(fmt.Sprintf("PR sidecar image: %s", sidecarImage))
-		for _, sidecar := range ta.maven.Spec.Sidecars {
-			if sidecar.Name != "proxy" {
-				continue
-			}
-			ta.Logf(fmt.Sprintf("Updating proxy sidecar with image %s", sidecarImage))
-			sidecar.Image = sidecarImage
-		}
-	}
-	ta.maven, err = tektonClient.TektonV1beta1().Tasks(ta.ns).Create(context.TODO(), ta.maven, metav1.CreateOptions{})
-	if err != nil {
-		debugAndFailTest(ta, err.Error())
-	}
-
-	pipelineYamlPath := filepath.Join(path, "..", "..", "hack", "examples", "pipeline.yaml")
-	ta.pipeline = &v1beta1.Pipeline{}
-	obj = streamFileYamlToTektonObj(pipelineYamlPath, ta.pipeline, ta)
-	ta.pipeline, ok = obj.(*v1beta1.Pipeline)
-	if !ok {
-		debugAndFailTest(ta, fmt.Sprintf("file %s did not produce a pipeline: %#v", pipelineYamlPath, obj))
-	}
-	ta.pipeline, err = tektonClient.TektonV1beta1().Pipelines(ta.ns).Create(context.TODO(), ta.pipeline, metav1.CreateOptions{})
-	if err != nil {
-		debugAndFailTest(ta, err.Error())
-	}
-
-	runYamlPath := filepath.Join(path, "..", "..", "hack", "examples", "run-e2e-shaded-app.yaml")
-	ta.run = &v1beta1.PipelineRun{}
-	obj = streamFileYamlToTektonObj(runYamlPath, ta.run, ta)
-	ta.run, ok = obj.(*v1beta1.PipelineRun)
-	if !ok {
-		debugAndFailTest(ta, fmt.Sprintf("file %s did not produce a pipelinerun: %#v", runYamlPath, obj))
-	}
-	ta.run, err = tektonClient.TektonV1beta1().PipelineRuns(ta.ns).Create(context.TODO(), ta.run, metav1.CreateOptions{})
-	if err != nil {
-		debugAndFailTest(ta, err.Error())
-	}
+	//test to make sure prow artifacts work the way I think the docs say they do
+	d1 := []byte("hello\ntest  artifacts\n")
+	_ = os.WriteFile(os.Getenv("ARTIFACTS")+"/TEST_ARTIFACT.txt", d1, 0644)
```

Comment:
@gabemontero based on the prow job authors guide (https://github.com/kubernetes/test-infra/blob/master/prow/jobs.md) I was expecting this to result in a 'TEST_ARTIFACT.txt' file I could download at the end. Any idea if this is possible?

I want to create a structured zip file of the results (build logs etc) with a HTML report that makes it easy to examine the current status.

Reply:
sorry @stuartwdouglas I have no experience with that specifically; though I *suspect* openshift CI is already overriding that to create the artifacts link you after you hit one of the Details links in the e2e status.

their docs on how they construct all that is at https://docs.ci.openshift.org/docs/how-tos/artifacts/

otherwise, their slack channel is #forum-testplatform on coreos slack

A couple of thoughts:
- there are ways to have specific tests results to show up in the junit section (I have seen it with the ginkgo based tests in openshift/origin), but I'm not sure how that gets wired up, or if it is even possible with our "in repo" form of testing.  I don't see me diving into trying to sort that out any time soon, but of course, if you are motivated, go for it.

- you can download an archive of all the gathered data today; for example, if you go to https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/redhat-appstudio_jvm-build-service/192/pull-ci-redhat-appstudio-jvm-build-service-main-jvm-build-service-in-repo-with-pruning-e2e/1560093510080663552/ and click on the download widget, you are provided with a gsutil invocation; I have been able to use that recently; check out https://cloud.google.com/storage/docs/gsutil_install.  You can also grab sub sections if you do down the dir structures

Reply:
I wanted to do something slightly different to both of those, basically take all the related failure information and group it in the same directory of a zip file, and then allow you to download that so you can easily peruse the failures.

I might have to take a different approach, and maybe look at making the way we dump logs attempt to group everything related.

Reply:
Ah, I see you have already done that at: https://github.com/redhat-appstudio/jvm-build-service/pull/196